### ISSUE TITLE
angular monorepo updates failing with npm 7

### ISSUE THREAD

**Issue by cyrilletuzi** [renovatebot/renovate#9561]
**What Renovate type, platform and version are you using?**

Renovate on GitHub.

**Describe the bug**

I'm the author of an Angular library, and since I updated my repo to npm 7 / `package-lock.json` version 2, `angular-monorepo` updates from Renovate are systematically failing because of versions mismatch.

Indeed, all `@angular/***` packages need to be in the exact same version, and npm 7 is stricter while checking `peerDependencies`.

You can see an example in cyrilletuzi/angular-async-local-storage#628.

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

### :warning: Artifact update problem

Renovate failed to update an artifact related to this branch. You probably do not want to merge this PR as-is.

:recycle: Renovate will retry this branch, including artifacts, only when one of the following happens:

 - any of the package files in this branch needs updating, or 
 - the branch becomes conflicted, or
 - you check the rebase/retry checkbox if found above, or
 - you rename this PR's title to start with "rebase!" to trigger it manually

The artifact failure details are included below:

##### File name: package-lock.json

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: @angular/common@11.2.9
npm ERR! node_modules/@angular/common
npm ERR!   @angular/common@"11.2.10" from the root project
npm ERR!   peer @angular/common@"11.2.9" from @angular/platform-browser@11.2.9
npm ERR!   node_modules/@angular/platform-browser
npm ERR!     @angular/platform-browser@"11.2.10" from the root project
npm ERR!     peer @angular/platform-browser@"11.2.9" from @angular/platform-browser-dynamic@11.2.9
npm ERR!     node_modules/@angular/platform-browser-dynamic
npm ERR!       @angular/platform-browser-dynamic@"11.2.10" from the root project
npm ERR!     1 more (@angular/router)
npm ERR!   2 more (@angular/platform-browser-dynamic, @angular/router)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! @angular/common@"11.2.10" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @angular/core@11.2.10
npm ERR! node_modules/@angular/core
npm ERR!   peer @angular/core@"11.2.10" from @angular/common@11.2.10
npm ERR!   node_modules/@angular/common
npm ERR!     @angular/common@"11.2.10" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /tmp/renovate-cache/others/npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/renovate-cache/others/npm/_logs/2021-04-15T03_07_16_624Z-debug.log

```

</details>

**Have you created a minimal reproduction repository?**

See cyrilletuzi/angular-async-local-storage#628

- [ ] I have provided a minimal reproduction repository
- [x] It happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

I fix the issue by redoing a full install on the branch (`rm -rf node_modules && rm package-lock.json && npm install`).

I guess the issue is a problem of order: packages (like `@angular/platform-browser`) relying on other (like `@angular/core`) must be updated *after*.

If getting the order right is too complex, using `--force` as suggested by the npm log may be a solution.

**Comment by rarkins** [comment]
I need more info to understand what exactly is going wrong here. According to the [first commit Renovate made](https://github.com/cyrilletuzi/angular-async-local-storage/pull/628/commits/436108df5b96f1fd61ce8018ff0ecf7361faa02c), every `@angular/*` lib seems to be updated to the same version:

![image](https://user-images.githubusercontent.com/6311784/114829287-2b946700-9dcb-11eb-89f8-1220cc0a910f.png)

But, looking at [your second commit](https://github.com/cyrilletuzi/angular-async-local-storage/pull/628/commits/b457b8e45138d0e994c79984c98da70252662b2f) it seems npm@7 has not updated the `dependencies` field of the `package-lock.json` when we ran `npm install`.

In other words it seems like npm's mistake, but our problem. I've seen multiple problems like this in npm@7 which lead me to believe it's still not very  robust

**Comment by cyrilletuzi** [comment]
Hi @rarkins, thanks for your quick answer and the amazing work.

Indeed, as indicated by the log I copied, it's the npm installation that fails.

How is Renovate proceeding in monorepo updates? Is it just redoing a `npm install` after updating `package.json` or is it updating each package one at a time?

I'll try to narrow the npm behavior based on this.

**Comment by rarkins** [comment]
Correction: `@angular/cli` is still at 11.2.9, although that's not something we can do anything about.

Here's the full error when running `npm install` after a manual `package.json` change:

<details>
<summary>npm install</summary>

```
# npm resolution error report

2021-04-15T07:23:54.653Z

Found: @angular/common@11.2.9
node_modules/@angular/common
  @angular/common@"11.2.10" from the root project
  peer @angular/common@"11.2.9" from @angular/platform-browser@11.2.9
  node_modules/@angular/platform-browser
    @angular/platform-browser@"11.2.10" from the root project
    peer @angular/platform-browser@"11.2.9" from @angular/platform-browser-dynamic@11.2.9
    node_modules/@angular/platform-browser-dynamic
      @angular/platform-browser-dynamic@"11.2.10" from the root project
    peer @angular/platform-browser@"11.2.9" from @angular/router@11.2.9
    node_modules/@angular/router
      @angular/router@"11.2.10" from the root project
  peer @angular/common@"11.2.9" from @angular/platform-browser-dynamic@11.2.9
  node_modules/@angular/platform-browser-dynamic
    @angular/platform-browser-dynamic@"11.2.10" from the root project
  peer @angular/common@"11.2.9" from @angular/router@11.2.9
  node_modules/@angular/router
    @angular/router@"11.2.10" from the root project

Could not resolve dependency:
@angular/common@"11.2.10" from the root project

Conflicting peer dependency: @angular/core@11.2.10
node_modules/@angular/core
  peer @angular/core@"11.2.10" from @angular/common@11.2.10
  node_modules/@angular/common
    @angular/common@"11.2.10" from the root project

Fix the upstream dependency conflict, or retry
this command with --force, or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.

Raw JSON explanation object:

{
  "code": "ERESOLVE",
  "current": {
    "name": "@angular/common",
    "version": "11.2.9",
    "location": "node_modules/@angular/common",
    "dependents": [
      {
        "type": "prod",
        "name": "@angular/common",
        "spec": "11.2.10",
        "error": "INVALID",
        "from": {
          "location": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
        }
      },
      {
        "type": "peer",
        "name": "@angular/common",
        "spec": "11.2.9",
        "from": {
          "name": "@angular/platform-browser",
          "version": "11.2.9",
          "location": "node_modules/@angular/platform-browser",
          "dependents": [
            {
              "type": "prod",
              "name": "@angular/platform-browser",
              "spec": "11.2.10",
              "error": "INVALID",
              "from": {
                "location": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
              }
            },
            {
              "type": "peer",
              "name": "@angular/platform-browser",
              "spec": "11.2.9",
              "from": {
                "name": "@angular/platform-browser-dynamic",
                "version": "11.2.9",
                "location": "node_modules/@angular/platform-browser-dynamic",
                "dependents": [
                  {
                    "type": "prod",
                    "name": "@angular/platform-browser-dynamic",
                    "spec": "11.2.10",
                    "error": "INVALID",
                    "from": {
                      "location": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
                    }
                  }
                ]
              }
            },
            {
              "type": "peer",
              "name": "@angular/platform-browser",
              "spec": "11.2.9",
              "from": {
                "name": "@angular/router",
                "version": "11.2.9",
                "location": "node_modules/@angular/router",
                "dependents": [
                  {
                    "type": "prod",
                    "name": "@angular/router",
                    "spec": "11.2.10",
                    "error": "INVALID",
                    "from": {
                      "location": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      {
        "type": "peer",
        "name": "@angular/common",
        "spec": "11.2.9",
        "from": {
          "name": "@angular/platform-browser-dynamic",
          "version": "11.2.9",
          "location": "node_modules/@angular/platform-browser-dynamic",
          "dependents": [
            {
              "type": "prod",
              "name": "@angular/platform-browser-dynamic",
              "spec": "11.2.10",
              "error": "INVALID",
              "from": {
                "location": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
              }
            }
          ]
        }
      },
      {
        "type": "peer",
        "name": "@angular/common",
        "spec": "11.2.9",
        "from": {
          "name": "@angular/router",
          "version": "11.2.9",
          "location": "node_modules/@angular/router",
          "dependents": [
            {
              "type": "prod",
              "name": "@angular/router",
              "spec": "11.2.10",
              "error": "INVALID",
              "from": {
                "location": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
              }
            }
          ]
        }
      }
    ]
  },
  "currentEdge": null,
  "edge": {
    "type": "prod",
    "name": "@angular/common",
    "spec": "11.2.10",
    "error": "INVALID",
    "from": {
      "location": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
    }
  },
  "peerConflict": {
    "name": "@angular/core",
    "version": "11.2.10",
    "whileInstalling": {
      "name": "@ngx-pwa/local-storage",
      "version": "11.1.0",
      "path": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
    },
    "location": "node_modules/@angular/core",
    "dependents": [
      {
        "type": "peer",
        "name": "@angular/core",
        "spec": "11.2.10",
        "from": {
          "name": "@angular/common",
          "version": "11.2.10",
          "whileInstalling": {
            "name": "@ngx-pwa/local-storage",
            "version": "11.1.0",
            "path": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
          },
          "location": "node_modules/@angular/common",
          "dependents": [
            {
              "type": "prod",
              "name": "@angular/common",
              "spec": "11.2.10",
              "from": {
                "location": "/Users/rhys/src/cyrilletuzi/angular-async-local-storage"
              }
            }
          ]
        }
      }
    ]
  },
  "strictPeerDeps": false,
  "force": false
}
```

</details>

**Comment by rarkins** [comment]
> How is Renovate proceeding in monorepo updates? Is it just redoing a `npm install` after updating `package.json` or is it updating each package one at a time?

Renovate updates the `package.json` manually then runs `npm install`. Until now this approach has worked for for npm/yarn/pnpm but npm@7 seems to be lacking in the same logic to validate/update lock files as was previously present.

**Comment by cyrilletuzi** [comment]
> Correction: @angular/cli is still at 11.2.9

It's normal, `@angular/cli` (which should not be confused with `@angular/compiler-cli`) is part of another monorepo (`angularcli`) in Renovate, which is correct because `@angular/cli` and `@angular/core` are not directly related and not always at the same exact version, so the issue is not about that.

> Renovate updates the package.json manually then runs npm install. Until now this approach has worked for for npm/yarn/pnpm but npm@7 seems to be lacking in the same logic to validate/update lock files as was previously present.

I'll try to narrow down the issue on the npm side, to see what would be the better solution (for example `npm install --force`) or if it is really a bug that should be reported directly to the npm team.

**Comment by rarkins** [comment]
Confirmed that when I delete `package-lock.json`, as well as `node_modules`, and then run `npm install` then it proceeds without conflict.

**Comment by rarkins** [comment]
Here's an interesting observation:
- I check out your project's master branch
- Update package.json versions manually to 11.2.10 except @angular/cli
- I delete package-lock.json manually but keep node_modules
- I run npm install and it produces an error

So it seems that the *existing* contents of `node_modules` is playing a part (which seems quite wrong to me.. they are not the source of truth here).

Additionally, it still fails if node_modules is empty but package-lock.json is there.

It also fails if node_modules is empty and the dependencies/devDependencies section of package-lock.json is updated manually before running install.

The only way I can find to update things right now is to wipe both node_modules and package-lock.json and run "npm install".

**Comment by rarkins** [comment]
Next test:

1. Check out master branch
2. Remove node_modules and package-lock.json
3. Run npm install
4. Commit the updated package-lock.json
5. Remove node_modules and package-lock.json
6. Update package.json versions manually
7. Run npm install (succeeds)

Here's the diff:

```diff
diff --git a/package-lock.json b/package-lock.json
index 815bca2..91e3134 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -8,12 +8,12 @@
       "name": "@ngx-pwa/local-storage",
       "version": "11.1.0",
       "dependencies": {
-        "@angular/common": "11.2.9",
-        "@angular/compiler": "11.2.9",
-        "@angular/core": "11.2.9",
-        "@angular/platform-browser": "11.2.9",
-        "@angular/platform-browser-dynamic": "11.2.9",
-        "@angular/router": "11.2.9",
+        "@angular/common": "11.2.10",
+        "@angular/compiler": "11.2.10",
+        "@angular/core": "11.2.10",
+        "@angular/platform-browser": "11.2.10",
+        "@angular/platform-browser-dynamic": "11.2.10",
+        "@angular/router": "11.2.10",
         "rxjs": "6.6.7",
         "tslib": "2.2.0"
       },
@@ -25,7 +25,7 @@
         "@angular-eslint/schematics": "2.1.0",
         "@angular-eslint/template-parser": "2.1.0",
         "@angular/cli": "11.2.9",
-        "@angular/compiler-cli": "11.2.9",
+        "@angular/compiler-cli": "11.2.10",
         "@types/jasmine": "3.6.9",
         "@types/node": "14.14.39",
         "@typescript-eslint/eslint-plugin": "4.22.0",
@@ -472,29 +472,29 @@
       }
     },
     "node_modules/@angular/common": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/common/-/common-11.2.9.tgz",
-      "integrity": "sha512-D60zF3F9pSpKmLKyYOjceREL2OKKIO20bZJzVqcgWQXUv0xxzHg9e6o2oOphPEj7kY/tKjIQjndRIBGA6MQHzg==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/common/-/common-11.2.10.tgz",
+      "integrity": "sha512-eguQEg5GQW1geUa9w00Nabz1BZKOj3rR1Wm6pYlLRF7KdWMDvpJ9xkG2P0X7YoqnWxX8GPqcsl2J3Ds+KDXJLA==",
       "dependencies": {
         "tslib": "^2.0.0"
       },
       "peerDependencies": {
-        "@angular/core": "11.2.9",
+        "@angular/core": "11.2.10",
         "rxjs": "^6.5.3"
       }
     },
     "node_modules/@angular/compiler": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/compiler/-/compiler-11.2.9.tgz",
-      "integrity": "sha512-EelccCc6Xgrz6UzoJU5EqrRc0CGDHS82cheiBP3lSDfjG6borD9al4DN5lfphp+FCmTpLY33wQ3jqhPwC02FMQ==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/compiler/-/compiler-11.2.10.tgz",
+      "integrity": "sha512-+Ba/y0v5eIu1fEJthIpFK+DiIT/YTYSL7Jy1hhZrGLHDiAUxSnzF5/lgrnLAPuZWt/5w44+WPXx3d1xqvNWWNw==",
       "dependencies": {
         "tslib": "^2.0.0"
       }
     },
     "node_modules/@angular/compiler-cli": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/compiler-cli/-/compiler-cli-11.2.9.tgz",
-      "integrity": "sha512-Gmd0pn1DCGfjQW2IWn4UbL7A/DtFEHBPGsP/3Mb7OfERs60NOra/HWAPDVfzcrnZqNRJBafrBvnNA8+1Ci5kNA==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/compiler-cli/-/compiler-cli-11.2.10.tgz",
+      "integrity": "sha512-IrCH/5VxFVCwklQQU+bhAQUuwr5nc0Ts7eId5CUyBh3edCdP6vnQaDi7yiJ3yxIo/DAwS68Yel2T1/2KAxp5gg==",
       "dev": true,
       "dependencies": {
         "@babel/core": "^7.8.6",
@@ -523,7 +523,7 @@
         "node": ">=10.0"
       },
       "peerDependencies": {
-        "@angular/compiler": "11.2.9",
+        "@angular/compiler": "11.2.10",
         "typescript": ">=4.0 <4.2"
       }
     },
@@ -546,9 +546,9 @@
       }
     },
     "node_modules/@angular/core": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/core/-/core-11.2.9.tgz",
-      "integrity": "sha512-OLW8ThHdG5XjE9PMdBWyAm19wXEjuhWAB19qCqNerKkk1RGEOn+ISUR5Rpg5Fh3rlI4nFOm+WeOUIaq9uxF53A==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/core/-/core-11.2.10.tgz",
+      "integrity": "sha512-1nZGSVZPvTA6rFJhoivcUKm2YBx9A+569cV1UTPlXjcpO+wNcrrXG+6yjxS2wAffTWEQ7xA0ajyo8mZrpqqw9Q==",
       "dependencies": {
         "tslib": "^2.0.0"
       },
@@ -558,16 +558,16 @@
       }
     },
     "node_modules/@angular/platform-browser": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/platform-browser/-/platform-browser-11.2.9.tgz",
-      "integrity": "sha512-a4sMSuSRWXNP4umhXrESFS6H5uc5CQ5c4VJMWpqVO5RK5FI9wKdU1Nyds2nZkQGkD5r6pOyGQeJvEbA+vq1SbQ==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/platform-browser/-/platform-browser-11.2.10.tgz",
+      "integrity": "sha512-8viTrC6W0FWzGou0cLHQLnXtTcXdm3V4fQWYcc03SiJM7sM/EDrBdwDqXEBUG089xrgbX8iDb+Txk821lyCa8w==",
       "dependencies": {
         "tslib": "^2.0.0"
       },
       "peerDependencies": {
-        "@angular/animations": "11.2.9",
-        "@angular/common": "11.2.9",
-        "@angular/core": "11.2.9"
+        "@angular/animations": "11.2.10",
+        "@angular/common": "11.2.10",
+        "@angular/core": "11.2.10"
       },
       "peerDependenciesMeta": {
         "@angular/animations": {
@@ -576,30 +576,30 @@
       }
     },
     "node_modules/@angular/platform-browser-dynamic": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/platform-browser-dynamic/-/platform-browser-dynamic-11.2.9.tgz",
-      "integrity": "sha512-6PawcsN5GqlQ4uILpdC748twODz34quYcCvF88wpNKsHhoxe/RPuf3m+XmbEPvsORVXLM/eOfjXcCrBfK06WPw==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/platform-browser-dynamic/-/platform-browser-dynamic-11.2.10.tgz",
+      "integrity": "sha512-IehLiP3UakVb+A44riNe8I/OQjDHMbCzhV9SupoUDP2Z7JEvrdoQFm9ijiu2hZTLPNWpwQog2b5iHytCbMW9ww==",
       "dependencies": {
         "tslib": "^2.0.0"
       },
       "peerDependencies": {
-        "@angular/common": "11.2.9",
-        "@angular/compiler": "11.2.9",
-        "@angular/core": "11.2.9",
-        "@angular/platform-browser": "11.2.9"
+        "@angular/common": "11.2.10",
+        "@angular/compiler": "11.2.10",
+        "@angular/core": "11.2.10",
+        "@angular/platform-browser": "11.2.10"
       }
     },
     "node_modules/@angular/router": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/router/-/router-11.2.9.tgz",
-      "integrity": "sha512-DNMaLXleQ4aaNJmVuwnKRkATjR3akUt4kjFFELiHMH8jEO2GBUveqsgjYibXsZ/5L6jSsvaxYvt0n2HGwxG9Tg==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/router/-/router-11.2.10.tgz",
+      "integrity": "sha512-OYVErGuC4ECWHcnYw3hmy34YKJG5BpH0uzhj/DxmvHEhDv4LHxSye5IcApktqDn68AvPmbGSmyjB1rNnXCsbAg==",
       "dependencies": {
         "tslib": "^2.0.0"
       },
       "peerDependencies": {
-        "@angular/common": "11.2.9",
-        "@angular/core": "11.2.9",
-        "@angular/platform-browser": "11.2.9",
+        "@angular/common": "11.2.10",
+        "@angular/core": "11.2.10",
+        "@angular/platform-browser": "11.2.10",
         "rxjs": "^6.5.3"
       }
     },
@@ -21648,25 +21648,25 @@
       }
     },
     "@angular/common": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/common/-/common-11.2.9.tgz",
-      "integrity": "sha512-D60zF3F9pSpKmLKyYOjceREL2OKKIO20bZJzVqcgWQXUv0xxzHg9e6o2oOphPEj7kY/tKjIQjndRIBGA6MQHzg==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/common/-/common-11.2.10.tgz",
+      "integrity": "sha512-eguQEg5GQW1geUa9w00Nabz1BZKOj3rR1Wm6pYlLRF7KdWMDvpJ9xkG2P0X7YoqnWxX8GPqcsl2J3Ds+KDXJLA==",
       "requires": {
         "tslib": "^2.0.0"
       }
     },
     "@angular/compiler": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/compiler/-/compiler-11.2.9.tgz",
-      "integrity": "sha512-EelccCc6Xgrz6UzoJU5EqrRc0CGDHS82cheiBP3lSDfjG6borD9al4DN5lfphp+FCmTpLY33wQ3jqhPwC02FMQ==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/compiler/-/compiler-11.2.10.tgz",
+      "integrity": "sha512-+Ba/y0v5eIu1fEJthIpFK+DiIT/YTYSL7Jy1hhZrGLHDiAUxSnzF5/lgrnLAPuZWt/5w44+WPXx3d1xqvNWWNw==",
       "requires": {
         "tslib": "^2.0.0"
       }
     },
     "@angular/compiler-cli": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/compiler-cli/-/compiler-cli-11.2.9.tgz",
-      "integrity": "sha512-Gmd0pn1DCGfjQW2IWn4UbL7A/DtFEHBPGsP/3Mb7OfERs60NOra/HWAPDVfzcrnZqNRJBafrBvnNA8+1Ci5kNA==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/compiler-cli/-/compiler-cli-11.2.10.tgz",
+      "integrity": "sha512-IrCH/5VxFVCwklQQU+bhAQUuwr5nc0Ts7eId5CUyBh3edCdP6vnQaDi7yiJ3yxIo/DAwS68Yel2T1/2KAxp5gg==",
       "dev": true,
       "requires": {
         "@babel/core": "^7.8.6",
@@ -21701,33 +21701,33 @@
       }
     },
     "@angular/core": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/core/-/core-11.2.9.tgz",
-      "integrity": "sha512-OLW8ThHdG5XjE9PMdBWyAm19wXEjuhWAB19qCqNerKkk1RGEOn+ISUR5Rpg5Fh3rlI4nFOm+WeOUIaq9uxF53A==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/core/-/core-11.2.10.tgz",
+      "integrity": "sha512-1nZGSVZPvTA6rFJhoivcUKm2YBx9A+569cV1UTPlXjcpO+wNcrrXG+6yjxS2wAffTWEQ7xA0ajyo8mZrpqqw9Q==",
       "requires": {
         "tslib": "^2.0.0"
       }
     },
     "@angular/platform-browser": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/platform-browser/-/platform-browser-11.2.9.tgz",
-      "integrity": "sha512-a4sMSuSRWXNP4umhXrESFS6H5uc5CQ5c4VJMWpqVO5RK5FI9wKdU1Nyds2nZkQGkD5r6pOyGQeJvEbA+vq1SbQ==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/platform-browser/-/platform-browser-11.2.10.tgz",
+      "integrity": "sha512-8viTrC6W0FWzGou0cLHQLnXtTcXdm3V4fQWYcc03SiJM7sM/EDrBdwDqXEBUG089xrgbX8iDb+Txk821lyCa8w==",
       "requires": {
         "tslib": "^2.0.0"
       }
     },
     "@angular/platform-browser-dynamic": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/platform-browser-dynamic/-/platform-browser-dynamic-11.2.9.tgz",
-      "integrity": "sha512-6PawcsN5GqlQ4uILpdC748twODz34quYcCvF88wpNKsHhoxe/RPuf3m+XmbEPvsORVXLM/eOfjXcCrBfK06WPw==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/platform-browser-dynamic/-/platform-browser-dynamic-11.2.10.tgz",
+      "integrity": "sha512-IehLiP3UakVb+A44riNe8I/OQjDHMbCzhV9SupoUDP2Z7JEvrdoQFm9ijiu2hZTLPNWpwQog2b5iHytCbMW9ww==",
       "requires": {
         "tslib": "^2.0.0"
       }
     },
     "@angular/router": {
-      "version": "11.2.9",
-      "resolved": "https://registry.npmjs.org/@angular/router/-/router-11.2.9.tgz",
-      "integrity": "sha512-DNMaLXleQ4aaNJmVuwnKRkATjR3akUt4kjFFELiHMH8jEO2GBUveqsgjYibXsZ/5L6jSsvaxYvt0n2HGwxG9Tg==",
+      "version": "11.2.10",
+      "resolved": "https://registry.npmjs.org/@angular/router/-/router-11.2.10.tgz",
+      "integrity": "sha512-OYVErGuC4ECWHcnYw3hmy34YKJG5BpH0uzhj/DxmvHEhDv4LHxSye5IcApktqDn68AvPmbGSmyjB1rNnXCsbAg==",
       "requires": {
         "tslib": "^2.0.0"
       }=
```

**Comment by rarkins** [comment]
This diff is minimal/simple and I don't see why npm is choking on it when the package-lock.json is present

**Comment by rarkins** [comment]
`npm i --force` maybe is ok. It warns but succeeds, and then running `npm i` after it succeeds without warnings.

**Comment by rarkins** [comment]
Here's my provisional idea:
- We need to run with `--force` for npm@7
- We could run a regular `npm i` after `npm i --force` so that any *permanent* errors are logged and reported as artifact errors.

**Comment by cyrilletuzi** [comment]
Which exact version of `npm` is Renovate using?

Because I've done the same test (installing Angular with 11.2.9, manually update `package.json` with 11.2.10 and run `npm install`) and it seems to work with npm v7.9.0 (a lot of issues regarding `peerDependencies` have been fixed in last versions).

**Comment by cyrilletuzi** [comment]
Sorry I may have done the test too quickly, I'll check again.

**Comment by cyrilletuzi** [comment]
The issue is still here with very last npm v7.9.0, sorry for the error.

**Comment by rarkins** [comment]
FYI I was testing against 7.8.0 locally. Renovate should take the latest available `npm` whenever it runs

**Comment by rarkins** [comment]
The problem summarized:
- `npm install` errors for no good reason
- `npm install --force` overcomes that, but it might also suppress *real* errors

**Comment by cyrilletuzi** [comment]
> npm install errors for no good reason

I've managed to narrow down the npm issue. I'll open an issue in npm repo and link it here, but I'll probably need some backup so it is taken into account (you seem better than me at explaining npm stuff).

**Comment by rarkins** [comment]
Great! I was thinking it should be raised as an issue there, but the main thing holding me back was that ideally we should be presenting them with a minimal reproduction (e.g. maybe even 2 dependencies could trigger this) and not a large repo example. Happy to follow and contribute to any issue you raise.

**Comment by cyrilletuzi** [comment]
Npm issue opened: npm/cli#3083

Could be good if you comment (you'll raise more attention than me, and that was quite difficult to explain clearly).

**Comment by HonkingGoose** [comment]
The npm bug report/issue (https://github.com/npm/cli/issues/3083) is now resolved with new code in the `npm/arborist` repository. I don't know when that code lands in a production `npm` 7 release.

Do we still need to do something on the Renovate end of things, or is this strictly a npm 7 bug?

**Comment by viceice** [comment]
I think this is a npm bug, hopefully they release a fixed version soon

**Comment by HonkingGoose** [comment]
I'm removing the `reproduction:needed` label, as I think we already have the full picture due to all the debugging here and in the upstream npm bug issue.

By removing the label I'm also stopping the stale bot from bothering us.

Feel free to re-add the label if necessary later on. 😄

**Comment by cyrilletuzi** [comment]
Both npm and Angular have weekly releases, so by the end of next week we should see in my repo if it's solved, and I'll close the issue.

**Comment by rarkins** [comment]
Arborist is a transitive dependency of npm so we wed to make sure we don't somehow end up with the old one too Eg due to lock file

**Comment by cyrilletuzi** [comment]
Update: the arborist PR didn't fix the issue, so I've opened a new issue, which has been flag "Priority 1": npm/cli#3171

**Comment by cyrilletuzi** [comment]
Since a few minor updates of npm, the initial error disappeared, but then there was another one (a falsely missing subdependency).

With the very last Angular 12.2.0, the Renovate PR successfully merged automatically, so the issue seems to be finally gone!

Thanks a lot to the Renovate team for their help and reactivity. 👍🏻
